Containers/mastercontainer/Dockerfile

```
@@ -56,6 +56,8 @@ RUN set -ex; \
     sed -i 's/^pm = dynamic/pm = ondemand/' /usr/local/etc/php-fpm.d/www.conf; \
     sed -i 's/^pm.max_children =.*/pm.max_children = 80/' /usr/local/etc/php-fpm.d/www.conf; \
     sed -i 's|access.log = /proc/self/fd/2|access.log = /proc/self/fd/1|' /usr/local/etc/php-fpm.d/docker.conf; \
+    grep -q ';listen.allowed_clients' /usr/local/etc/php-fpm.d/www.conf; \
+    sed -i 's|;listen.allowed_clients.*|listen.allowed_clients = 127.0.0.1,::1|' /usr/local/etc/php-fpm.d/www.conf; \
```

```suggestion
    grep -q 'listen =' /usr/local/etc/php-fpm.d/www.conf; \
    sed -i 's|listen =.*|listen = /var/run/php.sock|' /usr/local/etc/php-fpm.d/www.conf; \
```

And change the Apache PHP handler to: `SetHandler "proxy:unix:/var/run/php.sock"`

---

all right, lets do it like this then

---

https://github.com/docker-library/php/blob/e9e47fd185d6b36e491c713811437b0c17944faa/8.2/alpine3.18/fpm/Dockerfile#L249

```suggestion
    grep -q 'listen =' /usr/local/etc/php-fpm.d/www.conf; \
    sed -i 's|listen =.*|;listen = /var/run/php.sock # handled in zz-docker.conf|' /usr/local/etc/php-fpm.d/www.conf; \
    grep -q 'listen =' /usr/local/etc/php-fpm.d/zz-docker.conf; \
    sed -i 's|listen =.*|listen = /var/run/php.sock|' /usr/local/etc/php-fpm.d/zz-docker.conf; \
```

And change the Apache PHP handler to: `SetHandler "proxy:unix:/var/run/php.sock"`

---

healthcheck also needs to check if /var/run/php.sock exists

---

or we just actually limit it to 127.0.0.1:::1? would that be fine for you? Sounds like the easier approach...

---

due to the performance improvement?

---

https://serverfault.com/questions/124517/what-is-the-difference-between-unix-sockets-and-tcp-ip-sockets unix sockets also seems to be faster

---

I don't see why it is simpler to limit ips, instead of using a socket

---

2 lines of code vs 4? but yeah if you really want we can also use a socket

---

I would use the socket

---

@Zoey2936 lets switch to the socket in a follow-up.

I would welcome a PR :)